crates/typst-library/src/layout/transform.rs

```diff
@@ -109,15 +123,24 @@ impl Layout for RotateElem {
     ) -> SourceResult<Fragment> {
         let pod = Regions::one(regions.base(), Axes::splat(false));
         let mut frame = self.body().layout(vt, styles, pod)?.into_frame();
-        let Axes { x, y } = self
-            .origin(styles)
-            .resolve(styles)
-            .zip_map(frame.size(), FixedAlign::position);
+        let align = self.origin(styles).resolve(styles);
+        let Axes { x, y } = align.zip_map(frame.size(), FixedAlign::position);
+
         let ts = Transform::translate(x, y)
             .pre_concat(Transform::rotate(self.angle(styles)))
             .pre_concat(Transform::translate(-x, -y));
         frame.transform(ts);
-        Ok(Fragment::frame(frame))
+
+        // If we don't impact layout we exit early.
+        if !self.layout(styles) {
+            return Ok(Fragment::frame(frame));
+        }
+
+        // If we impact layout we wrap into a frame of the correct size
+        let (offset, size) = compute_bounding_box(&frame, ts);
+        let mut out = Frame::soft(size);
+        out.push(offset, FrameItem::Group(GroupItem::new(frame)));
```

Comment:
couldn't we transform the frame in place and avoid the extra nesting level? And even if not, `push_frame` is preferrable over creating a group item manually.

Reply:
This was not addressed.